api/clever/__init__.py

```diff
@@ -261,21 +260,20 @@ def remote_patron_lookup(self, token):
 
         identifier = data.get('id', None)
 
+        user_type = data.get('type', None)
+
         if not identifier:
             return INVALID_CREDENTIALS.detailed(lgt("A valid Clever login is required."))
 
-        if result.get('type') not in self.SUPPORTED_USER_TYPES:
+        if user_type not in self.SUPPORTED_USER_TYPES:
             return UNSUPPORTED_CLEVER_USER_TYPE
 
-        links = result['links']
-
-        user_link = [link for link in links if link['rel']
-                     == 'canonical'][0]['uri']
         # The canonical link includes the API version, so we use the base URL.
-        user = self._get(self.CLEVER_API_BASE_URL + user_link, bearer_headers)
+        user = self._get(self.CLEVER_API_VERSIONED_URL +
+                         'users/%s' % identifier, bearer_headers)
 
         user_data = user['data']
-        school_id = user_data['school']
+        school_id = user_data[user_type]['school']
```

Comment:
Is this `school_id` and the `student_grade` below nested inside a `roles` key? Looking at the [API Explorer](https://dev.clever.com/reference/getuser-1) for `/users/{id}` it looks like it might be?

Reply:
Yeah thanks for the catch... according to [this](https://dev.clever.com/docs/users#users-object-example) looks like userdata['roles'][user_type]['school'] is the proper nesting...